### PR TITLE
Resolved waring -Woverloaded-virtual on stitching module building

### LINKED ISSUE #27467: Proposal: ObjectDetection utility class for detection + annotation in OpenCV
### Describe the feature and motivation

 Proposal: Add a Utility Class for Object Detection Inference and Annotation

I propose to add a utility class named ObjectDetection under a new or existing module (e.g., objdetect) in OpenCV that simplifies the pipeline for:

Running object detection using a provided model

Annotating the prediction results on images or video

🎯 Key Features:
Accepts a pre-trained model (OpenCV cv2.dnn, ONNX, PyTorch, etc.)

Accepts image(s) or video stream as input

Automatically handles:

Inference

Drawing bounding boxes

Overlaying class labels and confidence scores

Outputs annotated image or video

🧩 Motivation:
Many users (especially beginners or prototyping engineers) perform object detection and then manually write repetitive code to draw results using OpenCV primitives like cv2.rectangle and cv2.putText. A clean abstraction that integrates model inference and result annotation would:

Improve productivity

Encourage best practices (e.g., consistent styling, confidence thresholding)

Reduce boilerplate code in object detection applications

🏗️ Possible Extension:
Built-in color mapping for labels

Optional saving/display of output

Confidence thresholding

Format-agnostic support (accepting detection outputs from YOLO, SSD, etc.)

I’d love to hear your thoughts on whether this feature aligns with OpenCV’s direction. If approved, I’m happy to begin working on a prototype and PR under guidance.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Priyadarshan-Garg**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work Fixes #27467
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by Kumataro** [comment]
Thank you for your PR for this issue, https://github.com/opencv/opencv/issues/27752 !

Append `CV_OVERRIDE` on `feed()` definition makes build errors.
So my PR, https://github.com/opencv/opencv/pull/27753,  suppress build warnings only

```bash
In file included from /build/precommit_linux64/4.x/opencv/modules/core/include/opencv2/core.hpp:52:0,
                 from /build/precommit_linux64/4.x/opencv/modules/stitching/src/precomp.hpp:57,
                 from /build/precommit_linux64/4.x/opencv/modules/stitching/src/exposure_compensate.cpp:43:
/build/precommit_linux64/4.x/opencv/modules/core/include/opencv2/core/cvdef.h:797:23: error: virt-specifiers in 'feed' not allowed outside a class definition
 #  define CV_OVERRIDE override
                       ^
/build/precommit_linux64/4.x/opencv/modules/stitching/src/exposure_compensate.cpp:465:80: note: in expansion of macro 'CV_OVERRIDE'
                              const std::vector<std::pair<UMat,uchar> > &masks) CV_OVERRIDE
```

File: modules/stitching/src/exposure_compensate.cpp [hunk 1]
```diff
@@ -460,8 +460,9 @@ void ChannelsCompensator::setMatGains(std::vector<Mat>& umv)
 
 
 template<class Compensator>
+
```
Comment (asmorkalov) [inline modules/stitching/src/exposure_compensate.cpp:463]: Extra empty line is not needed here.

File: modules/stitching/src/exposure_compensate.cpp [hunk 2]
```diff
@@ -460,8 +460,9 @@ void ChannelsCompensator::setMatGains(std::vector<Mat>& umv)
...
 void BlocksCompensator::feed(const std::vector<Point> &corners, const std::vector<UMat> &images,
-                             const std::vector<std::pair<UMat,uchar> > &masks)
+                             const std::vector<std::pair<UMat,uchar> > &masks) CV_OVERRIDE
```
Comment (asmorkalov) [inline modules/stitching/src/exposure_compensate.cpp:465]: CV_OVERRIDE should be added in header file, but not implementation,

Comment (Priyadarshan-Garg) [reply modules/stitching/src/exposure_compensate.cpp:465]: To be honest I wanted to contribute to this repository so badly that I messed up. It was hard for me to find the actual cause of that error so I took help from Gemini and I failed. I'm really 😔.

**Comment by asmorkalov** [comment]
I investigated the issue in deep and looks like @Kumataro is right. It's not override and cannot be fixed by modifiers. I proposed alternative solution in https://github.com/opencv/opencv/pull/27758
